presto-main/src/main/java/io/prestosql/sql/planner/PlanBuilder.java

```diff
@@ -85,28 +93,40 @@ public TranslationMap getTranslations()
         return translations;
     }
 
+    public Scope getScope()
+    {
+        return translations.getScope();
+    }
+
     public PlanBuilder appendProjections(Iterable<Expression> expressions, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
     {
-        TranslationMap translations = copyTranslations();
+        return appendProjections(expressions, symbolAllocator, idAllocator, TranslationMap::rewrite, TranslationMap::canTranslate);
+    }
 
+    public <T extends Expression> PlanBuilder appendProjections(
```

Comment:
where is this specialized method used (apart from projection above)?

Reply:
In planGroupingOperations